storage/log_storage.go

```diff
@@ -119,6 +119,10 @@ type LeafReader interface {
 	// same hash but different sequence numbers. If orderBySequence is true then the returned data
 	// will be in ascending sequence number order.
 	GetLeavesByHash(leafHashes [][]byte, orderBySequence bool) ([]*trillian.LogLeaf, error)
+	// GetLeavesByLeafIdentityHash returns leaves by their identity hash. Note: This is for
+	// internal use only and is not part of the public API. The returned leaf objects may not
+	// be fully populated
```

Comment:
I haven't confirmed but I think I can probably move this to the wrapper. It's only for use in handling duplicate submissions. It can't fully populate the leaf struct because they might not have been sequenced yet.